mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3089 @@
+#         kr17kev_in_hz = guess*(bins[1]-bins[0])+bins[0]
+        #self.B_field = B(17.8, kr17kev_in_hz + 0)
+        if self.resolution_function == 'simulated_resolution':
+            if self.fixed_scatter_proportion == True:
+                self.results = self.fit_data_ftc(freq_bins, data_hist_freq)
+            else:
+                self.results = self.fit_data_ftc_2(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'gaussian_resolution':
+            if self.fixed_scatter_proportion == True:
+                self.results = self.fit_data_1(freq_bins, data_hist_freq)
+            else:
+                self.results = self.fit_data(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'gaussian_lorentzian_composite_resolution':
+            if self.fixed_scatter_proportion == True and self.fixed_survival_probability == True:
+                self.results = self.fit_data_composite_gaussian_lorentzian_fixed_scatter_proportion_and_survival_prob(freq_bins, data_hist_freq)
+            elif self.fixed_scatter_proportion == True and self.fixed_survival_probability == False:
+                self.results = self.fit_data_composite_gaussian_lorentzian_fixed_scatter_proportion(freq_bins, data_hist_freq)
+            elif self.fixed_scatter_proportion == False and self.fixed_survival_probability == True and self.partially_fixed_scatter_proportion == False:
+                self.results = self.fit_data_composite_gaussian_lorentzian_fixed_survival_probability(freq_bins, data_hist_freq)
+            elif self.partially_fixed_scatter_proportion == True and self.fixed_survival_probability == True:
+                self.results = self.fit_data_composite_gaussian_lorentzian_fixed_survival_probability_partially_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'elevated_gaussian':
+            self.results = self.fit_data_elevated_gaussian_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'composite_gaussian':
+            self.results = self.fit_data_composite_gaussian_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'composite_gaussian_pedestal_factor':
+            self.results = self.fit_data_composite_gaussian_pedestal_factor_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'composite_gaussian_scaled':
+            self.results = self.fit_data_composite_gaussian_scaled_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'simulated_resolution_scaled':
+            if self.fit_recon_eff == False:
+                self.results = self.fit_data_simulated_resolution_scaled_fixed_scatter_proportion(freq_bins, data_hist_freq)
+            else:
+                self.results = self.fit_data_simulated_resolution_scaled_fit_recon_eff(freq_bins, data_hist_freq)
```

Comment:
please add some description in comments so if-elif strcuture becomes more easily readable